mls-rs/src/group/external_commit.rs

```diff
@@ -171,33 +209,45 @@ impl<C: ClientConfig> ExternalCommitBuilder<C> {
         let psk_ids = self
             .external_psks
             .into_iter()
-            .map(|psk_id| {
-                Ok(PreSharedKeyID {
-                    key_id: JustPreSharedKeyID::External(psk_id),
-                    psk_nonce: PskNonce::random(&cipher_suite_provider)
-                        .map_err(|e| MlsError::CryptoProviderError(e.into_any_error()))?,
-                })
-            })
+            .map(|psk_id| PreSharedKeyID::new(JustPreSharedKeyID::External(psk_id), &cipher_suite))
             .collect::<Result<Vec<_>, MlsError>>()?;
 
-        let external_init = Proposal::ExternalInit(ExternalInit { kem_output });
+        let mut proposals = vec![Proposal::ExternalInit(ExternalInit { kem_output })];
 
         #[cfg(feature = "psk")]
-        let proposals = psk_ids
-            .into_iter()
-            .map(|psk| Proposal::Psk(PreSharedKeyProposal { psk }))
-            .chain([external_init]);
-
-        #[cfg(not(feature = "psk"))]
-        let proposals = [external_init].into_iter();
-
-        let proposals = proposals
-            .chain(self.to_remove.map(|r| {
-                Proposal::Remove(RemoveProposal {
-                    to_remove: LeafIndex(r),
-                })
-            }))
-            .collect::<Vec<_>>();
+        proposals.extend(
+            psk_ids
+                .into_iter()
+                .map(|psk| Proposal::Psk(PreSharedKeyProposal { psk })),
+        );
+
+        #[cfg(feature = "custom_proposal")]
+        {
+            let mut custom_proposals = self.custom_proposals;
+            proposals.append(&mut custom_proposals);
+        }
+
+        #[cfg(all(feature = "custom_proposal", feature = "by_ref_proposal"))]
+        for message in self.received_custom_proposals {
+            let MlsMessagePayload::Plain(plaintext) = message.payload else {
+                return Err(MlsError::UnexpectedMessageType);
+            };
+
+            let auth_content = AuthenticatedContent::from(plaintext.clone());
+
+            verify_plaintext_authentication(&cipher_suite, plaintext, None, None, &group.state)
```

Comment:
This wouldn't happen elsewhere in the code?

Reply:
No, this code basically replaces `verify_plaintext_authentication` from the one done by `Group` to the one done by `ExternalGroup` (skip MAC).

So `Group` calls (1) `get_event_from_incoming_message` and then (2) `process_event_or_content` (inside `process_incoming_message_with_time`). And (1) for proposals consists mostly of `verify_plaintext_authentication`. So here we replace (1) with weaker verification and leave (2) intact.